### ISSUE TITLE
🐛 useExhaustiveDependencies false positive on types

### ISSUE THREAD

**Issue by ABuffSeagull** [biomejs/biome#811]
### Environment information

```block
CLI:
  Version:                      1.3.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "tmux-256color"
  JS_RUNTIME_VERSION:           "v20.9.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

`useCallback` references the type of a components props through `typeof` and biome thinks it needs to be added to the hook's dependencies

https://biomejs.dev/playground/?bracketSpacing=false&lintRules=all&code=aQBtAHAAbwByAHQAIAB7AHUAcwBlAEMAYQBsAGwAYgBhAGMAawB9ACAAZgByAG8AbQAgACcAcgBlAGEAYwB0ACcAOwAKAAoAZgB1AG4AYwB0AGkAbwBuACAAQQBwAHAAKAB7AG8AdQB0AGUAcgB9ADoAIAB7AG8AdQB0AGUAcgA6ACAAcwB0AHIAaQBuAGcAWwBdAH0AKQAgAHsACgAgACAAdQBzAGUAQwBhAGwAbABiAGEAYwBrACgAKABpAG4AbgBlAHIAOgAgACgAdAB5AHAAZQBvAGYAIABvAHUAdABlAHIAKQBbAG4AdQBtAGIAZQByAF0AKQAgAD0APgAgAHsACgAgACAAIAAgAGkAbgBuAGUAcgAgACsAIAAxADsACgAgACAAfQAsACAAWwBdACkAOwAKAH0ACgBBAHAAcAAoACkAOwA%3D

### Expected result

Shouldn't be an error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
